PatternPal/PatternPal.Extension/Commands/SubscribeEvents.cs

```diff
@@ -249,6 +249,33 @@ internal static void OnFileDelete(object sender, FileSystemEventArgs fileSystemE
             LogEventResponse response = PushLog(request);
         }
 
+        /// <summary>
+        /// The event handler for handling the File.Rename Event. The file watcher detects every file renamed,
+        /// so any event triggers with files other than .cs files are unhandled.    
+        /// </summary>
+        /// <param name="sender"></param>
+        /// <param name="fileSystemEventArgs"></param>
+        internal static void OnFileRename(object sender, RenamedEventArgs e)
+        {
+            // Only log for the creation of .cs files
+            if (Path.GetExtension(e.OldName) != ".cs" || Path.GetExtension(e.Name) != ".cs")
```

Comment:
Is this really the proper check to perform here? What if you are renaming `example.txt` to example.cs`? The other way around feels less relevant, but I think we should think a bit more about this.